rclc_lifecycle/src/rclc_lifecycle/rclc_lifecycle.c

```diff
@@ -39,6 +43,8 @@ rclc_make_node_a_lifecycle_node(
   bool enable_communication_interface
 )
 {
+  lifecycle_node->node = node;
```

Comment:
check for NULL pointer for all arguments (and add test cases for them)
see example here: https://github.com/ros2/rclc/blob/e0252bdb7d86a01688d22d76cdbe6e3d960a6aec/rclc/src/rclc/node.c#L29